bitcoin/src/blockdata/locktime/relative.rs

```diff
@@ -38,6 +39,23 @@ pub enum LockTime {
 }
 
 impl LockTime {
+    /// Try to interpret the given number as a relative lock time.
+    #[inline]
+    pub fn from_num(num: i64) -> Option<LockTime> {
+        let int = u32::try_from(num).ok()?;
+
+        if int & Sequence::LOCK_TIME_DISABLE_FLAG_MASK != 0 {
+            return None;
+        }
+
+        let low16 = int as u16; // only need lowest 16 bits
+        if int & Sequence::LOCK_TYPE_MASK > 0 {
+            Some(LockTime::from(Time::from_512_second_intervals(low16)))
+        } else {
+            Some(LockTime::from(Height::from(low16)))
+        }
+    }
```

Comment:
This one we already have (for `u32`s) in #2549.

We could add a wrapper function for `i64`s since that's what you get from scriptnums, but I think it might make more sense to just directly add the ability to parse locktimes from Script.

Reply:
> I think it might make more sense to just directly add the ability to parse locktimes from Script.

Yes, please!

Reply:
cc #2535 where we talked about this (and I wanted to mention this issue but couldn't find it since it is titled "Single patch" lol).

Also cc #2849 which is the "followup to #2535" issue.